### PR TITLE
WIP: Rename General File Directory to Library-Specific File Directory

### PR DESCRIPTION
**Pull request by tomeg09**
Describe the changes you have made here: what, why, ..

Per issue https://github.com/koppor/jabref/issues/571, there was a request to rename "general file directory" to "Library-specific file directory". This has been updated in the following places:
1. Within the internal code: this has been renamed to "Library Specific File Directory" (or librarySpecificFileDirectory, LIBRARY_SPECIFIC_FILE_DIRECTORY, etc).
2. Within the test base: all associated tests where this has been updated have been changed (I have therefore declared that I "created tests" as the testing library may require PR approver's scrutiny.
3. A localisation for "Library-specific File Directory" was added for English. I have left the localisation for "General File Directory" in place. I have not added localisations for other languages.

Screenshots of the change reflected in the UI:
image

Documentation updates:
User Docs: see [Docs PR#535](https://github.com/JabRef/user-documentation/pull/535)
Developer Docs: [this file](https://docs.jabref.org/finding-sorting-and-cleaning-entries/filelinks#directories-for-files) makes reference to the general file directory and requires updating. 

Lines in question are:

> Library → Library properties, items General file directory, and User-specific file directory. (plus screenshot)
>
> JabRef uses these 3 directories to search for the files: JabRef starts in the user-specific file directory, then the general file directory, and, finally, the main file directory​
>
> Either, each user can set his directory in the "Main file directory". In case the group also shares papers and thus there are two directories (the private one and a group-shared one), one can set a directory within the library (the "General file directory")
>
> If no match is found, it uses the "General file directory" of the bib file. If that is not found, it uses the one configured at File → Preferences → Linked files.
>
> It is also possible to achieve this result by setting . as "General file directory" in the library properties.
>
>Relative file directories obviously only work in the library properties for a bib file, e.g. a.bib Library → Library properties → General file directory → papers.

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable) 
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/libraryproperties/general/GeneralPropertiesViewModel.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void setValues() {
 
         selectedEncodingProperty.setValue(metaData.getEncoding().orElse(StandardCharsets.UTF_8));
         selectedDatabaseModeProperty.setValue(metaData.getMode().orElse(BibDatabaseMode.BIBLATEX));
-        generalFileDirectoryProperty.setValue(metaData.getDefaultFileDirectory().orElse("").trim());
```
Comment (koppor) [inline src/main/java/org/jabref/gui/libraryproperties/general/GeneralPropertiesViewModel.java:62]: Rename the property in the UI code, too. Consistency!

File: src/main/java/org/jabref/model/database/BibDatabaseContext.java [hunk 2]
```diff
@@ -165,12 +165,12 @@ public List<Path> getFileDirectories(FilePreferences preferences) {
...
         userFileDirectory.ifPresent(fileDirs::add);
 
-        Optional<Path> generalFileDirectory = metaData.getDefaultFileDirectory().map(dir -> getFileDirectoryPath(dir));
+        Optional<Path> generalFileDirectory = metaData.getLibraryFileDirectory().map(dir -> getFileDirectoryPath(dir));
         generalFileDirectory.ifPresent(fileDirs::add);
 
         // fileDirs.isEmpty() is true after these two if there are no directories set in the BIB file itself:
         //   1) no user-specific file directory set (in the metadata of the bib file) and
-        //   2) no general file directory is set (in the metadata of the bib file)
+        //   2) no library file directory is set (in the metadata of the bib file)
```
Comment (koppor) [inline src/main/java/org/jabref/model/database/BibDatabaseContext.java:173]: When reading this,  I think. everywhere, the “-specific“ should be added. Also in constants, getters, ...

**Review by koppor — CHANGES_REQUESTED** [review]
I think the name of variables and UI text should be consistent.  Please add “specific“ where appropriate!

**Comment by koppor** [comment]
@tomeg09 Any chance to get my wish implemented soon? https://github.com/JabRef/jabref/pull/12059 needs this change!

**Comment by tomeg09** [comment]
> @tomeg09 Any chance to get my wish implemented soon? #12059 needs this change!

@koppor - working on this now

**Review by koppor — COMMENTED** [review]
Looks good! Can you add a CHANGELOG.md entry, too?

File: CHANGELOG.md [hunk 3]
```diff
@@ -65,6 +65,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We changed instances of 'Search Selected' to 'Search Pre-configured' in Web Search Preferences UI. [#11871](https://github.com/JabRef/jabref/pull/11871)
 - We added a new CSS style class `main-table` for the main table. [#11881](https://github.com/JabRef/jabref/pull/11881)
 - When renaming a file, the old extension is now used if there is none provided in the new name. [#11903](https://github.com/JabRef/jabref/issues/11903)
+- We changed the name of the library-based file directory from 'General File Directory' to 'Library-specific File Directory' per issue [#571](https://github.com/koppor/jabref/issues/571)
```
Comment (tomeg09) [inline CHANGELOG.md:68]: @koppor

Comment (koppor) [reply CHANGELOG.md:68]: Ah, my mobile client kind of hid it. Thank you for the quick note!

**Review by koppor — APPROVED** [review]

**Pull request merged**